app/thank-you/page.tsx

```diff
@@ -0,0 +1,32 @@
+'use client';
+
+import React, { useEffect, useState } from 'react';
+import { isSignInWithEmailLink } from 'firebase/auth';
+import { auth } from '../config/firebase';
+import { authenticateAndSaveUserDataFromEmailRedirect } from '../services/emailAuth';
+import { useRouter } from 'next/navigation';
+import Loading from '../loading';
+import ThankYou from '../components/main-content/ThankYou';
+
+export default function ThankYouPage() {
+  const [isLoading, setIsLoading] = useState(true);
+
+  const router = useRouter();
+  useEffect(() => {
+    if (!isSignInWithEmailLink(auth, window.location.href)) {
+      router.push('/');
+    } else {
+      let email = window.localStorage.getItem('emailForSignIn');
+      if (email) {
+        authenticateAndSaveUserDataFromEmailRedirect(email);
+        setIsLoading(false);
+      } else {
+        router.push('/');
```

Comment:
I saw you mentioned in the PR a toast is needed here to notify the user why they will be redirected to the signup page again. Could you ask Lucie to create a ticket for this too?